desc/objectives/_omnigenity.py

```diff
@@ -501,6 +502,232 @@ def compute(self, params, constants=None):
+        Type of omnigenity (M, N). Default = quasi-isodynamic (0, 1).
+    M_booz : int, optional
+        Poloidal resolution of Boozer transformation. Default = 2 * eq.M.
+    N_booz : int, optional
+        Toroidal resolution of Boozer transformation. Default = 2 * eq.N.
+    well_weight : float, optional
+        Weight applied to the bottom of the magnetic well (B_min) relative to the top
+        of the magnetic well (B_max). Default = 1, which weights all points equally.
+    name : str
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = False
+    _units = "(T)"
+    _print_value_fmt = "Omnigenity error: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        field,
+        target=0,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        grid_eq=None,
+        grid_field=None,
+        rho=1.0,
+        M_booz=None,
+        N_booz=None,
+        well_weight=1,
+        name="omnigenity",
+    ):
+        self._grid_eq = grid_eq
+        self._grid_field = grid_field
+        self._rho = rho
+        self.helicity = field.helicity
+        self.M_booz = M_booz
+        self.N_booz = N_booz
+        self.well_weight = well_weight
+        super().__init__(
+            things=[eq, field],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self.things[0]
+        field = self.things[1]
+        M_booz = self.M_booz or 2 * eq.M
+        N_booz = self.N_booz or 2 * eq.N
+
+        if self._grid_eq is None:
+            grid_eq = LinearGrid(
+                rho=self._rho, M=2 * M_booz, N=2 * N_booz, NFP=eq.NFP, sym=False
+            )
+        else:
+            grid_eq = self._grid_eq
+
+        if self._grid_field is None:
+            # TODO: is this a good default grid?
+            # oversamples in eta by a factor of 2, including points at B_min & B_max
+            # 8 field lines, including alpha=0 and alpha=pi
+            grid_field = LinearGrid(
+                rho=self._rho, theta=2 * field.M_well, zeta=8, NFP=field.NFP, sym=False
+            )
+        else:
+            grid_field = self._grid_field
+
+        self._dim_f = grid_field.num_nodes
+        self._eq_data_keys = ["|B|_mn"]
+        self._field_data_keys = ["|B|_omni", "h"]
+
+        assert grid_eq.NFP == grid_field.NFP
+        assert grid_eq.sym is False
```

Comment:
`assert not grid.sym`

Reply:
should also add error messages here, and possibly use `errorif` to raise a ValueError instead of assertion error